src/harness.js

```diff
@@ -550,6 +550,28 @@ class Harness {
   }
   set subject(value) { this.options.subject = value; }
 
+
+  /**
+   * `userSettingsDoc` from the {@link HarnessInputs} set through the constructor
+   * @default {Object} A constructed object of type `user-settings` https://docs.communityhealthtoolkit.org/core/overview/db-schema/#users based on
+   * known user information
+   */
+  get userSettingsDoc () {
+    if (this.options.userSettingsDoc) {
+      return this.options.userSettingsDoc;
+    }
+  
+    const user = this.user;
+    return {
+      _id: `org.couchdb.user:${user._id || 'unknown'}`,
+      name: user._id,
+      type: 'user-settings',
+      contact_id: user._id,
```

Comment:
Should we fail loudly if user._id is undefined? I see you are trying to work around that on [line 566](https://github.com/medic/medic-conf-test-harness/pull/111/files#diff-d356eda9f1d71e0f25fec3e869c8bcbfdbfd2cd258f70df3b3425188641ca33bR566)

Reply:
since this is in a "getter", i'm not comfortable throwing here.  so i can change the interface to be `harness.getUserSettingsDoc()` and `setUserSettingsDoc(doc)` and then fail fast. generally, i like the rule that getters should be "dumb" and "always work" but it is the classes' responsibility to [avoid invalid state](https://softwareengineering.stackexchange.com/questions/261585/should-a-getter-throw-an-exception-if-its-object-has-invalid-state)

it is definitely confusing to special-case the undefined _id on 566 and not 567/569. so let's definitely remove that

i'm not sure what to do exactly -- i'm added some unit tests to ensure we fail-fast in operations. unclear whether it is legit to bend the rules for a test library like this where invalid states are more normal.  here it is pretty easy - "no id" is never possible.

Reply:
>so since this is in a "getter", i'm not comfortable throwing here. so i can change the interface to be harness.getUserSettingsDoc() and setUserSettingsDoc(doc) and then fail fast. do you think that would be an improvement

I believe that would. On the flip side: is there value doing this? What amount of hair pulling can we avoid by failing loudly? If we don't ever expect this to be undefined, let's fail fast - at the constructor or setter stage

Reply:
Great feedback. But I feel the get/set interface for userSettingsDoc is pretty inconsistent with the other existing interfaces -- so let's think it through with an improvement @ https://github.com/medic/cht-conf-test-harness/issues/125